server/src/main/java/io/deephaven/server/session/SessionServiceGrpcImpl.java

```diff
@@ -186,6 +176,37 @@ public void exportFromTicket(
                 });
     }
 
+    @Override
+    public void publishFromTicket(
+            @NotNull final PublishRequest request,
+            @NotNull final StreamObserver<PublishResponse> responseObserver) {
+        final SessionState session = service.getCurrentSession();
+
+        if (!request.hasSourceId()) {
+            responseObserver
+                    .onError(Exceptions.statusRuntimeException(Code.INVALID_ARGUMENT, "Source ticket not supplied"));
+            return;
+        }
+        if (!request.hasResultId()) {
+            responseObserver
+                    .onError(Exceptions.statusRuntimeException(Code.INVALID_ARGUMENT, "Result ticket not supplied"));
+            return;
+        }
+
+        final SessionState.ExportObject<Object> source = ticketRouter.resolve(
+                session, request.getSourceId(), "sourceId");
+        Ticket resultId = request.getResultId();
+
+        final SessionState.ExportBuilder<Object> publisher = ticketRouter.publish(
+                session, resultId, "resultId", () -> {
+                    // when publish is complete, complete the gRPC request
+                    GrpcUtil.safelyComplete(responseObserver, PublishResponse.getDefaultInstance());
+                });
```

Comment:
In the ExportTicketResolver case, this safelyComplete is happening on thread as opposed to as part of a submit - is that correct?

Reply:
Yeah, but you are right that is kind of bogus; might as well submit a non-export.